src/components/headers/collection-header.tsx

```diff
@@ -503,6 +556,37 @@ export class CollectionHeader extends React.Component<IProps, IState> {
     );
   }
 
+  private checkUploadPrivilleges(collection) {
+    const addAlert = () => {
+      const alert = new AlertType();
```

Comment:
alerts without variant should not be valid,
and `new AlertType` is not needed...


```suggestion
      const alert = { title: t`You don't have rights to do this operation.`, variant: 'error' };
```

Reply:
But this causes error: Type '{ title: string; variant: string; }' is not assignable to type 'AlertType'
Also variant seems to be restricted to this: AlertType.variant: "default" | "success" | "danger" | "warning" | "info"

Reply:
oh, sorry, same issue .. if you change variant to one of those, the first error disappears as well

danger is the red one :)

Reply:
I tried yesterday and today again and still:
TS2322: Type '{ title: string; variant: string; }' is not assignable to type 'AlertType'.
ui_1           |     561 |       const alert = { title: t`You don't have rights to do this operation.`, variant: 'warning' };
ui_1           |     562 |       this.setState({
ui_1           |   > 563 |         alerts: [...this.state.alerts, alert],
ui_1           |         |                                        ^^^^^
ui_1           |     564 |       });

Reply:
ah, for some reason TS doesnt see the types as compatible when going through a variable... surprising :)

This worked for me...

```diff
diff --git a/src/components/headers/collection-header.tsx b/src/components/headers/collection-header.tsx
index d1c7ec78..f1fc2ccd 100644
--- a/src/components/headers/collection-header.tsx
+++ b/src/components/headers/collection-header.tsx
@@ -558,10 +558,14 @@ export class CollectionHeader extends React.Component<IProps, IState> {
 
   private checkUploadPrivilleges(collection) {
     const addAlert = () => {
-      const alert = new AlertType();
-      alert.title = t`You don't have rights to do this operation.`;
       this.setState({
-        alerts: [...this.state.alerts, alert],
+        alerts: [
+          ...this.state.alerts,
+          {
+            title: t`You don't have rights to do this operation.`,
+            variant: 'danger',
+          },
+        ],
       });
     };
 
@@ -909,10 +913,16 @@ export class CollectionHeader extends React.Component<IProps, IState> {
 
   private toggleImportModal(isOpen: boolean, warning?: string) {
     if (warning) {
-      const newAlert = new AlertType();
-      newAlert.title = warning;
-      newAlert.variant = 'warning';
-      this.setState({ alerts: [...this.state.alerts, newAlert] });
+      this.setState({
+        alerts: [
+          ...this.state.alerts,
+          {
+            title: warning,
+            variant: 'warning',
+          },
+        ],
+      });
     }
     this.setState({ showImportModal: isOpen });
   }
```

but also, this is an alternative...

```diff
diff --git a/src/components/headers/collection-header.tsx b/src/components/headers/collection-header.tsx
index d1c7ec78..75fb41ff 100644
--- a/src/components/headers/collection-header.tsx
+++ b/src/components/headers/collection-header.tsx
@@ -558,8 +558,10 @@ export class CollectionHeader extends React.Component<IProps, IState> {
 
   private checkUploadPrivilleges(collection) {
     const addAlert = () => {
-      const alert = new AlertType();
-      alert.title = t`You don't have rights to do this operation.`;
+      const alert: AlertType = {
+        title: t`You don't have rights to do this operation.`,
+        variant: 'danger',
+      };
       this.setState({
         alerts: [...this.state.alerts, alert],
       });
@@ -909,9 +911,10 @@ export class CollectionHeader extends React.Component<IProps, IState> {
 
   private toggleImportModal(isOpen: boolean, warning?: string) {
     if (warning) {
-      const newAlert = new AlertType();
-      newAlert.title = warning;
-      newAlert.variant = 'warning';
+      const newAlert: AlertType = {
+        title: warning,
+        variant: 'warning',
+      };
       this.setState({ alerts: [...this.state.alerts, newAlert] });
     }
     this.setState({ showImportModal: isOpen });
```

Reply:
Done.